gi_notifier/model/model.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass, field
+
+
+@dataclass
+class Resin:
+    """
+    Data class for resins.
+    """
+    original_resin: str = """Use your <a href="https://genshin-impact.fandom.com/wiki/Original_Resin">Original Resin</a> for"""
+    transient_resin: str = """Do not forget to claim your <a href="https://genshin-impact.fandom.com/wiki/Transient_Resin">Transient Resin</a> from <a href="https://genshin-impact.fandom.com/wiki/Serenitea_Pot/Realm_Depot">Realm Depot</a>"""
+    condensed_resin: str = """Making <a href="https://genshin-impact.fandom.com/wiki/Condensed_Resin">Condensed Resin</a> for future usage"""
+
+
+@dataclass
+class FixedMaterials:
+    """
+    Dataclass for the fixed materials.
+    """
+    experience: str = """Obtaining <a href="https://genshin-impact.fandom.com/wiki/Experience">Experience</a> from <a href="https://genshin-impact.fandom.com/wiki/Ley_Line_Outcrop">Ley Line Outcrop - Blossom of Revelation</a>"""
+    mora: str = """Obtaining <a href="https://genshin-impact.fandom.com/wiki/Mora">Mora</a> from <a href="https://genshin-impact.fandom.com/wiki/Ley_Line_Outcrop">Ley Line Outcrop - Blossom of Wealth</a>"""
+    resin: Resin = field(default_factory=Resin)
+
+
+@dataclass
+class Characters:
+    """
+    Data class for characters.
+    """
+    title: str
+    normal_boss_drop: str
+    weekly_boss_drop: str
+    talent_book: str
+    fixed_materials: FixedMaterials = field(default_factory=FixedMaterials)
```

Comment:
Can these fields be further reduced into two dataclasses?

Namely, for the `Item` and `Source` with them both having `name: str` and `link: str` as the data members. 

Should help with avoiding the repetition of the string like the following.

```
"""Obtaining <a href="https://genshin-impact.fandom.com/wiki/Ensnaring_Gaze">Ensnaring Gaze</a> from <a href="https://genshin-impact.fandom.com/wiki/Tenebrous_Papilla">Tenebrous Papilla</a>"""
```

Into the following

```python3
f"""Obtaining <a href="{skirk.normal_boss_drop.item.link}">{skirk.normal_boss_drop.item.name}</a> from <a href="{skirk.normal_boss_drop.source.link}">{skirk.normal_boss_drop.source.name}</a>"""
```

Let us focus more on the data and less on what is being shown.